### PR TITLE
Fearture sync process for shop

### PR DESCRIPTION
**Pull request by phong-jack**

### CODE REVIEW

**Comment by anvotnm** [comment]
Pretty good! but please update a little
Using a while loop for synchronization is fine, but let's improve it a bit.

You should create another Queue (I will call it Sync_Queue or whatever you want) for this synchronization
The process starts when you call an API that adds a job to the Sync_Queue with jobData:
{
   page: 1,
   count: 100
}

Then you write a consumer (processor)  for which this job will fetch data and make changes to your shop, etc...
if jobData.page * jobDatacount < reply.data.Total then add another job into Sync_Queue which is:
{
   page: page +1,
   count: 100
} 

Do this if your app crashes during sync, it will be able to continue the sync

**Comment by voxuanan** [comment]
Pretty good! but please update a little
Using a while loop for synchronization is fine, but let's improve it a bit.

You should create another Queue (I will call it Sync_Queue or whatever you want) for this synchronization
The process starts when you call an API that adds a job to the Sync_Queue with jobData:
{
page: 1,
count: 100
}

Then you write a consumer (processor) for which this job will fetch data and make changes to your shop, etc...
if jobData.page * jobDatacount < reply.data.Total then add another job into Sync_Queue which is:
{
page: page +1,
count: 100
}

Do this if your app crashes during sync, it will be able to continue the sync

**Comment by voxuanan** [comment]
Remember to assign yourself and request me as reviewers

File: src/common/fetch/services/fetch.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { ShopService } from 'src/modules/shop/services/shop.service';
+
+@Injectable()
+export class FetchService {
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly shopService: ShopService,
+  ) {}
+
+  async fetchShopData(): Promise<void> {
```
Comment (voxuanan) [inline src/common/fetch/services/fetch.service.ts:15]: What I mentioned above is here

**Review by voxuanan — CHANGES_REQUESTED** [review]

**Comment by voxuanan** [comment]
LGTM!

**Pull request merged**